main.cpp

```diff
@@ -311,6 +241,12 @@ void updateValue( T& value)
     value += value;
 }
 
+template <typename T>
+void cube( std::unique_ptr<T>&(value) )
```

Comment:
why do you have `( )` around `value`?   It's confusing to look at and figure out the purpose of it.
- [ ] revise